Documentation/ApplyingCorePatches/Index.rst

```diff
@@ -0,0 +1,225 @@
+   Automatic application of patches with this method only works with
+   Composer. If you still run a legacy TYPO3 installation consider to
+   :ref:`migrate to Composer <migratetocomposer>`.
+
+To automatically apply patches first install `cweagans/composer-patches`:
+
+.. code-block:: bash
+
+   composer req cweagans/composer-patches
+
+Chose a folder to store all patches in. This folder should ideally lie outside
+of the webroot. Here we use the folder :file:`patches` on the same level like
+the projects main :file:`composer.json`. Each patch can be applied to exactly
+one composer package. The paths used in the patch must be relative to the
+packages path.
+
+Edit your projects main :file:`composer.json`. Add a section :json:`patches`
+within the section :json:`extra`. If there is no section :json:`extra` yet
+add one.
+
+.. code-block:: json
+   :lineos: 5-9
+   :caption: project_root/composer.json
+
+   "extra": {
+     "typo3/cms": {
+       "web-dir": "public"
+     },
+     "patches": {
+       "typo3/cms-core": {
+         "Bug #98106 fix something":"patches/Bug-98106.diff"
+       }
+     }
+   }
+
+The patch itself can look like this:
+
+.. code-block:: none
+   :caption: patches/Bug-98106.diff (Simplified)
+
+   diff --git a/Classes/Utility/GeneralUtility.php b/Classes/Utility/GeneralUtility.php
+   index be47cfe..08fd6fc 100644
+   --- a/Classes/Utility/GeneralUtility.php
+   +++ b/Classes/Utility/GeneralUtility.php
+   @@ -2282,17 +2282,24 @@
+         */
+        public static function createVersionNumberedFilename($file)
+        {
+   +        $isFrontend = ($GLOBALS['TYPO3_REQUEST'] ?? null) instanceof ServerRequestInterface
+   +            && ApplicationType::fromRequest($GLOBALS['TYPO3_REQUEST'])->isFrontend();
+            $lookupFile = explode('?', $file);
+            $path = $lookupFile[0];
+
+   -        if (($GLOBALS['TYPO3_REQUEST'] ?? null) instanceof ServerRequestInterface
+   -            && ApplicationType::fromRequest($GLOBALS['TYPO3_REQUEST'])->isFrontend()
+   -        ) {
+   +        if ($isFrontend) {
+                $mode = strtolower($GLOBALS['TYPO3_CONF_VARS']['FE']['versionNumberInFilename']);
+                if ($mode === 'embed') {
+                    $mode = true;
+
+
+Apply the patch by running the following command:
+
+.. code-block:: bash
+
+   composer install
+
+.. note::
+   Changes to the patch file after the patch was applied successfully once will
+   not be automatically applied. In that case delete the installed sources
+   then execute :bash:`composer install` once more.
+
+If applying the patch fails you will often get a cryptic error message like:
+
+.. code-block:: none
+   Example error message
+
+   Could not apply patch! Skipping. The error was: Cannot apply patch patches/Bug-98106.diff
+
+You can get a more verbose error message by calling:
+
+.. code-block:: bash
+   :caption: Very, very verbose error message
+
+   composer install -vvv
+
+Creating a diff from a Core change
```

Comment:
We should also mention https://packagist.org/packages/gilbertsoft/typo3-core-patches in this chapter, as this greatly simplifies applying core patches.

Reply:
I never used it. Would you like to add a chapter, @brotkrueml ?

Reply:
Done